### PR TITLE
Theme and icons

### PR DESCRIPTION
**Pull request by mvkv**

### CODE REVIEW

File: components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useReducer } from 'react';
+import { Moon, Sun } from 'react-feather';
+import { Theme } from '../../types/themes';
+import { DEFAULT_THEME, themeReducer } from '../../store/theme';
+
+export default function Navbar() {
+  const [theme, dispatchTheme] = useReducer(themeReducer, DEFAULT_THEME);
+
+  const handleSetTheme = () => {
+    const newTheme = { newTheme: theme.currentTheme === Theme.DARK ? Theme.LIGHT : Theme.DARK };
```
Comment (powahftw) [inline components/Navbar/Navbar.tsx:10]: nit: no reason to assign to a variable, also you just need dispatch({...}) no reason to return anything

File: components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return dispatchTheme(newTheme);
+  };
+
+  return (
+    <div className="flex justify-between w-full p-4">
+      <div> </div>
```
Comment (powahftw) [inline components/Navbar/Navbar.tsx:16]: Avoid empty divs

File: components/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      <div>
+        <button type="button" onClick={handleSetTheme}>{theme.currentTheme === Theme.LIGHT ? <Moon /> : <Sun color="white" />}</button>
```
Comment (powahftw) [inline components/Navbar/Navbar.tsx:18]: nit: Nest them so it's easier to see the difference between attributes and the button content

File: components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  const handleSetTheme = () => {
+    const newTheme = { newTheme: theme.currentTheme === Theme.DARK ? Theme.LIGHT : Theme.DARK };
+    return dispatchTheme(newTheme);
+  };
+
+  return (
+    <div className="flex justify-between w-full p-4">
+      <div> </div>
+      <div>
```
Comment (powahftw) [inline components/Navbar/Navbar.tsx:17]: Do you need this wrapper div?

File: components/PillButton/PillButton.tsx [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-                 transition-transform duration-200 transform
-                 hover:translate-y-px focus:translate-y-px active:translate-y-1.5
-               bg-blue-100 hover:bg-blue-200 text-blue-500 border-b-4 border-blue-500
-                 py-2 px-4"
+      className="flex flex-row items-center text-blue-500 hover:text-blue-600
+                px-4 py-2 hover:bg-blue-200
+                bg-blue-100 rounded-full shadow-md border-solid border-b-4 border-blue-500
+                transition-transform duration-400 transform
+                hover:translate-y-px focus:translate-y-px active:translate-y-1.5"
+      onClick={onClickAction}
```
Comment (powahftw) [inline components/PillButton/PillButton.tsx:10]: nit: keep this at the start to cut down on unecessary diff

File: pages/_app.css [hunk 6]
```diff
@@ -0,0 +1,8 @@
+:root {
+    --bg-color: #ffffff;
+}
+
+body {
+    background: var(--bg-color);
+    transition: background-color 200ms linear;
+}
```
Comment (powahftw) [inline pages/_app.css:8]: missing newline

File: pages/index.tsx [hunk 7]
```diff
@@ -30,7 +32,7 @@ export default function Home() {
   // Handles creating timer on play
   useEffect(() => {
     if (state.timerState !== TimerState.PLAYING) {
-      return;
+      return undefined;
```
Comment (powahftw) [inline pages/index.tsx:35]: disable this check and remove this, i'm not a fan of this return

File: store/theme.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { Theme } from '../types/themes';
+
+export type State = {
+  currentTheme: Theme;
+}
+
+export type Action = {
+  newTheme: Theme;
+}
```
Comment (powahftw) [inline store/theme.ts:9]: no reason to make this convoluted (yet)

I'd just create an action called change theme and keep here the transition logic (if it's dark it goes to light and the other way around)

File: store/theme.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+const themeToColor = new Map<Theme, string>([
+  [Theme.DARK, '#232629'],
+  [Theme.LIGHT, '#e3f1ff'],
+]);
+
+export function themeReducer(state: State, action: Action) {
+  const bgColor = themeToColor.get(action.newTheme);
+  document.documentElement.style.setProperty('--bg-color', bgColor);
```
Comment (powahftw) [inline store/theme.ts:22]: We should not have sideEffect here. 

https://stackoverflow.com/questions/47833824/side-effects-in-redux-reducer

We should have a useEffect listener near the root that takes care of upading the document style prop when the theme change, but it should not be in the reducer.

File: types/themes.ts [hunk 10]
```diff
@@ -0,0 +1,4 @@
+export enum Theme {
+    DARK,
+    LIGHT
+}
```
Comment (powahftw) [inline types/themes.ts:4]: missing empty lines

File: components/Navbar/Navbar.tsx (hunk 1 above)
Comment (mvkv) [reply components/Navbar/Navbar.tsx:10]: Done

File: store/theme.ts (hunk 9 above)
Comment (mvkv) [reply store/theme.ts]: Done, ended up using Context to pass theme and his dispatcher down to the tree, put useEffect into index page to handle theme updating.

**Pull request merged**